nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -269,17 +274,58 @@ BtNavigator::navigateToPose()
 
       // action server feedback (pose, duration of task,
       // number of recoveries, and distance remaining to goal)
+      geometry_msgs::msg::PoseStamped current_pose;
       nav2_util::getCurrentPose(
-        feedback_msg->current_pose, *tf_, global_frame_, robot_frame_, transform_tolerance_);
+        current_pose, *tf_, global_frame_, robot_frame_, transform_tolerance_);
 
       geometry_msgs::msg::PoseStamped goal_pose;
       blackboard_->get("goal", goal_pose);
 
-      feedback_msg->distance_remaining = nav2_util::geometry_utils::euclidean_distance(
-        feedback_msg->current_pose.pose, goal_pose.pose);
+      // Get current path points
+      nav_msgs::msg::Path current_path;
+      blackboard_->get("path", current_path);
+
+      // Find the closest pose to current pose on global path
+      auto find_closest_pose_idx = 
+        [&current_pose, &current_path]() {
+          size_t closest_pose_idx = 0;
+          double curr_min_dist = std::numeric_limits<double>::max();
+          for (auto curr_it = current_path.poses.begin(); curr_it != current_path.poses.end(); ++curr_it) {
+            double curr_dist = nav2_util::geometry_utils::euclidean_distance(current_pose, *curr_it);
+            if (curr_dist < curr_min_dist) {
+              curr_min_dist = curr_dist;
+              closest_pose_idx = curr_it - current_path.poses.begin();
+            }
+          }
+          return closest_pose_idx;
+        };
+
+      size_t closest_pose_idx = find_closest_pose_idx();
+
+      // Calculate distance on the path
+      double distance_remaining =
+        nav2_util::geometry_utils::calculate_path_length(current_path, closest_pose_idx);
```

Comment:
```suggestion
        nav2_util::geometry_utils::calculate_path_length(current_path, find_closest_pose_idx());
```

This lets you remove L303 `size_t closest_pose_idx = find_closest_pose_idx();` and the white space around it